src/services/findAllReferences.ts

```diff
@@ -2090,6 +2090,9 @@ namespace ts.FindAllReferences {
                 cancellationToken.throwIfCancellationRequested();
                 return mapDefined(getPossibleSymbolReferenceNodes(sourceFile, node.text), ref => {
                     if (isStringLiteralLike(ref) && ref.text === node.text) {
+                        if (ref.flags & (NodeFlags.JSDoc & ~NodeFlags.JavaScriptFile)) {

```

Comment:
Again, isn't `NodeFlags.JSDoc & ~NodeFlags.JavaScriptFile` just `NodeFlags.JSDoc`?

Reply:
Should search be allowed in JsDoc tags in JavaScript files?

Reply:
> Second, I personally think that trying to defend against JSDoc all around the language service is not going to be the right approach because there will always be holes

Currently, I see the problem when we try to get a type of TypeLiteral at JsDoc in TypeScript files. I'm not sure where the best way to handle that...

Reply:
@DanielRosenwasser Can I close this and https://github.com/microsoft/TypeScript/pull/51057 PRs?

Reply:
Yes, I think we should close these and solve the larger problem in https://github.com/microsoft/TypeScript/issues/51104 instead (probably by binding jsdoc everywhere and fixing any crashes that are exposed.)

Reply:
Oke, do I need to revers these changes https://github.com/microsoft/TypeScript/pull/50757?

Reply:
Mmmmm...yeah, I think so. @DanielRosenwasser do you agree?